pkg/phlaredb/block_querier.go

```diff
@@ -905,6 +938,75 @@ func (b *singleBlockQuerier) SelectMatchingProfiles(ctx context.Context, params
 	return iter.NewMergeIterator(maxBlockProfile, false, iters...), nil
 }
 
+func (b *singleBlockQuerier) Series(ctx context.Context, params *ingestv1.SeriesRequest) ([]*typesv1.Labels, error) {
+	sp, ctx := opentracing.StartSpanFromContext(ctx, "Series Block")
+	defer sp.Finish()
+
+	// TODO(bryan) do we need to close this?
+	err := b.Open(ctx)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
`Open` is very expensive operation, therefore and we do want to keep blocks open. Unfortunately, at the moment we don't have a straight-forward caching / eviction policy, which may easily become a problem. We will change this in the future

Reply:
Potentially we also what it to have a state where only the TSDB index is there and parquet files are not even looked at.

Reply:
It makes sense we try keep things open for as long as possible if opens are expensive. I like @simonswine's idea of doing a partial open vs loading the entire block.